src/latch_cli/services/launch/launch_v2.py

```diff
@@ -5,17 +5,92 @@
 
 import dill
 import google.protobuf.json_format as gpjson
+import gql
 from flyteidl.core import interface_pb2 as _interface_pb2
 from flytekit.core.context_manager import FlyteContextManager
 from flytekit.core.promise import translate_inputs_to_literals
 from flytekit.core.type_engine import TypeTransformerFailedError
-from flytekit.models.interface import Variable, VariableMap
+from flytekit.models.interface import Parameter, ParameterMap, Variable, VariableMap
 
 from latch.utils import current_workspace
 from latch_cli import tinyrequests
 from latch_cli.services.launch.interface import get_workflow_interface
 from latch_cli.utils import get_auth_header
 from latch_sdk_config.latch import config
+from latch_sdk_gql.execute import execute
+
+
+def launch_from_launch_plan(*, wf_name: str, lp_name: str, version: Optional[str] = None) -> int:
+    target_account_id = current_workspace()
+
+    wf_id, interface, default_params = get_workflow_interface(target_account_id, wf_name, version)
+    default_params_map: dict[str, Parameter] = ParameterMap.from_flyte_idl(gpjson.ParseDict(default_params, _interface_pb2.ParameterMap())).parameters
+    parameter_interface: dict[str, Variable] = VariableMap.from_flyte_idl(gpjson.ParseDict(interface, _interface_pb2.VariableMap())).variables
+
+    query_lp_default = gql.gql(
+        """
+        query LaunchPlanDefaultInputs($workflowId: BigInt!, $namePattern: String!) {
+            lpInfos(
+                filter: {
+                    workflowId: { equalTo: $workflowId },
+                    name: { like: $namePattern }
+                },
+                first: 1
```

Comment:
hmm this will be ambiguous in the case that a person has two launchplans with the same name in different files, probably fine to just fail if there is more than one result here for now and add a todo to allow filtering by fqn of the launchplan too